src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -147,9 +147,13 @@ public static function createConnection($dsn, array $options = [])
         }
 
         if (isset($params['host']) || isset($params['path'])) {
-            if (!isset($params['dbindex']) && isset($params['path']) && preg_match('#/(\d+)$#', $params['path'], $m)) {
-                $params['dbindex'] = $m[1];
-                $params['path'] = substr($params['path'], 0, -\strlen($m[0]));
+            if (!isset($params['dbindex']) && isset($params['path'])) {
+                if (preg_match('#/(\d+)$#', $params['path'], $m)) {
+                    $params['dbindex'] = $m[1];
+                    $params['path'] = substr($params['path'], 0, -\strlen($m[0]));
+                } else {
+                    throw new InvalidArgumentException(sprintf('Invalid Redis DSN: "%s".', $dsn));
```

Comment:
Could it give an hint about what is actually wrong with the given DSN? Just asking, really

Reply:
Good point. What do you think of this?
 throw new InvalidArgumentException(sprintf('Invalid Redis DSN: "%s", database index  should be number.', $dsn));

Reply:
looks good, here is a proposal that is consistent with what we do elsewhere
```suggestion
                    throw new InvalidArgumentException(sprintf('Invalid Redis DSN: "%s", The "dbindex" parameter must be 
 a number.', $dsn));
```